src/core/ext/xds/xds_cluster.cc

```diff
@@ -645,6 +646,34 @@ absl::StatusOr<std::shared_ptr<const XdsClusterResource>> CdsResourceParse(
     cds_update->override_host_statuses.Add(
         XdsHealthStatus(XdsHealthStatus::kHealthy));
   }
+  // Record telemetry labels (if any).
+  cds_update->telemetry_labels =
+      std::make_shared<std::map<std::string, std::string>>();
+  const envoy_config_core_v3_Metadata* metadata =
+      envoy_config_cluster_v3_Cluster_metadata(cluster);
+  if (metadata != nullptr) {
+    google_protobuf_Struct* telemetry_labels;
+    if (envoy_config_core_v3_Metadata_filter_metadata_get(
+            metadata,
+            StdStringToUpbString(
+                absl::string_view("com.google.csm.telemetry_labels")),
+            &telemetry_labels)) {
+      size_t iter = -1;
```

Comment:
probably not a big deal, but you don't need to initialize `iter` and `fields_entry`